### ISSUE TITLE
Mlton needs to be patched

### ISSUE THREAD

**Issue by tel** [copumpkin/nixpkgs#20]
```
installing `mlton-20130715'
these derivations will be built:
  /nix/store/milmjcfhs1sh247jcas0p8s1ii7yqabr-mlton-20130715.drv
building path(s) `/nix/store/lykk9zilvkpamx0xrpwm3s9bgz17xjp8-mlton-20130715'
building /nix/store/lykk9zilvkpamx0xrpwm3s9bgz17xjp8-mlton-20130715
unpacking sources
unpacking source archive /nix/store/6sp047h3j9m4nanyvhbmwyga59w8shi9-mlton-20130715-1.amd64-darwin.gmp-macports.tgz
unpacking source archive /nix/store/802i6iwbpn9xj3nh64i1w6z5vnnwc0p2-mlton-20130715.src.tgz
source root is mlton-20130715
patching sources
configuring
building
make flags: SHELL=/nix/store/x2lgxlqd3mgx2504rcbhjnix2b695qja-bash-4.3-p30/bin/bash all-no-docs
building all-no-docs
make dirs runtime compiler basis-no-check script mlbpathmap constants libraries tools
make[1]: Entering directory `/private/var/folders/lj/655x29_s66718hvw6bdcvy1w0000gn/T/nix-build-mlton-20130715.drv-0/mlton-20130715'
building dirs
mkdir -p "/private/var/folders/lj/655x29_s66718hvw6bdcvy1w0000gn/T/nix-build-mlton-20130715.drv-0/mlton-20130715/build/bin" "/private/var/folders/lj/655x29_s66718hvw6bdcvy1w0000gn/T/nix-build-mlton-20130715.drv-0/mlton-20130715/build/lib/include"
mkdir -p "/private/var/folders/lj/655x29_s66718hvw6bdcvy1w0000gn/T/nix-build-mlton-20130715.drv-0/mlton-20130715/build/lib/targets/self/include"
mkdir -p "/private/var/folders/lj/655x29_s66718hvw6bdcvy1w0000gn/T/nix-build-mlton-20130715.drv-0/mlton-20130715/build/lib/targets/self/sml"
building runtime
Compiling MLton runtime system for self.
make -C runtime
make[2]: Entering directory `/private/var/folders/lj/655x29_s66718hvw6bdcvy1w0000gn/T/nix-build-mlton-20130715.drv-0/mlton-20130715/runtime'
building gdtoa/README
gzip -dc gdtoa.tgz | tar xf -
patch -s -p0 <gdtoa.may_alias-unions.patch
patch -s -p0 <gdtoa.rename-public-fns.patch
patch -s -p0 <gdtoa.hide-private-fns.patch
patch -s -p0 <gdtoa.hide-public-fns.patch
building gdtoa/dmisc.c
building gdtoa/arithchk.out
cd gdtoa && gcc -std=gnu99   -fno-common  -pedantic -Wall -W -Wextra -Wformat=2 -Wswitch-default -Wswitch-enum -Wuninitialized -Winit-self -Wstrict-aliasing=2 -Wfloat-equal -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wwrite-strings -Waggregate-return -Wstrict-prototypes -Wold-style-definition -Wmissing-prototypes -Wmissing-declarations -Wmissing-field-initializers -Wmissing-noreturn -Wmissing-format-attribute -Wredundant-decls -Wnested-externs -fvisibility=hidden -m64 -I/usr/local/include -I/sw/include -I/opt/local/include -I. -Iplatform -Wdisabled-optimization -O2 -fomit-frame-pointer -w -O1 -o arithchk.out arithchk.c
/nix/store/x2lgxlqd3mgx2504rcbhjnix2b695qja-bash-4.3-p30/bin/bash: gcc: command not found
make[2]: *** [gdtoa/arithchk.out] Error 127
make[2]: Leaving directory `/private/var/folders/lj/655x29_s66718hvw6bdcvy1w0000gn/T/nix-build-mlton-20130715.drv-0/mlton-20130715/runtime'
make[1]: *** [runtime] Error 2
make[1]: Leaving directory `/private/var/folders/lj/655x29_s66718hvw6bdcvy1w0000gn/T/nix-build-mlton-20130715.drv-0/mlton-20130715'
make: *** [all-no-docs] Error 2
builder for `/nix/store/milmjcfhs1sh247jcas0p8s1ii7yqabr-mlton-20130715.drv' failed with exit code 2
error: build of `/nix/store/milmjcfhs1sh247jcas0p8s1ii7yqabr-mlton-20130715.drv' failed
```

**Comment by tel** [comment]
SMLNJ is also broken, but perhaps it's not relevant here.

**Comment by gridaphobe** [comment]
Ah right, mlton was fixed in 43c0939f4409c4771bc6c77596aaa937dc6f2b03
